src/ring_theory/graded_algebra/basic.lean

```diff
@@ -140,3 +139,45 @@ begin
 end
 
 end graded_algebra
+
+section homogeneous_element
+open_locale pointwise
+
+variables {ι R A : Type*}
+variables [comm_semiring R] [ring A] [algebra R A]
+variables (𝒜 : ι → submodule R A)
+
+/-- An element `a : A` is said to be homogeneous if there is some
+`i : ι` such that `a ∈ 𝒜 i`-/
+def is_homogeneous (a : A) : Prop := ∃ i, a ∈ 𝒜 i
```

Comment:
These typeclasses are way stronger than they need to be. You can state this about `graded_monoid`s:
```
/-- An element `a : A` is said to be homogeneous if there is some
`i : ι` such that `a ∈ 𝒜 i`-/
def is_homogeneous (a : A) : Prop := ∃ i, a ∈ 𝒜 i

variables {𝒜}
lemma is_homogenous.mul [has_add ι] [has_mul A] [set_like.has_graded_mul 𝒜] {a b : A} :
  is_homogeneous 𝒜 a → is_homogeneous 𝒜 b → is_homogeneous 𝒜 (a * b)
| ⟨i, hi⟩ ⟨j, hj⟩ := ⟨_, set_like.has_graded_mul.mul_mem hi hj⟩
variables (𝒜)

lemma is_homogenous_one [has_zero ι] [has_one A] [set_like.has_graded_one 𝒜] :
  is_homogeneous 𝒜 (1 : A) := ⟨_, set_like.has_graded_one.one_mem⟩

/-- When `𝒜` is a `set_like.graded_monoid`, then the elements satisfying `is_homogenous` form a submonoid. --/
def homogenous_submonoid [add_comm_monoid ι] [monoid A] [set_like.graded_monoid 𝒜] :
  submonoid A :=
{ carrier := {a | is_homogeneous 𝒜 a},
  one_mem' := is_homogenous_one 𝒜,
  mul_mem' := λ a b, is_homogenous.mul }
```

Reply:
(and so these probably belong in `algebra/graded_monoid.lean`)